icu4c/source/test/intltest/rbbitst.cpp

```diff
@@ -4276,6 +4277,101 @@ void RBBITest::TestBug5532(void)  {
 }
 
 
+void RBBITest::TestJpParticleAndPunctuation(void)  {
+    //Expected breaking result:［京都▁観光］▁時雨▁殿に▁行った。
+
+    const char16_t *s16 = {u"［京都観光］時雨殿に行った。"};
```

Comment:
These test cases can be added to the test file `source/test/testdata/rbbitst.txt`, without the need to write any code.